File: haskell/lucians-luscious-lasagna/src/LuciansLusciousLasagna.hs
module LuciansLusciousLasagna (elapsedTimeInMinutes, expectedMinutesInOven, preparationTimeInMinutes) where

expectedMinutesInOven :: Int
expectedMinutesInOven = 40

preparationTimeInMinutes :: Int -> Int
preparationTimeInMinutes layers = layers * 2

elapsedTimeInMinutes :: Int -> Int -> Int
elapsedTimeInMinutes layers elapsed = elapsed + preparationTimeInMinutes layers

File: fsharp/lucians-luscious-lasagna/LuciansLusciousLasagna.fs
module LuciansLusciousLasagna

let expectedMinutesInOven = 40

let remainingMinutesInOven x = expectedMinutesInOven - x

let preparationTimeInMinutes layers = 2 * layers

let elapsedTimeInMinutes layers elapsed =
    (preparationTimeInMinutes layers) + elapsed
